[DOC-2554] Security section (#627)
harness/developer-hub: docs/chaos-engineering/overview/harness-chaos-security.md

Both of the above-mentioned actions require specific access to the infrastructure (or service) in question. This information is expected to be fed as Kubernetes secrets to the transient chaos pod resources which are launched during experiment execution. Refer [here](https://developer.harness.io/docs/chaos-engineering/technical-reference/architecture) to learn more about how experiments are executed. 

**Note:** The experiment artifact (stored in the ChaosHubs or provided as input by the user during experiment creation) only references the secret names, with the lifecycle of actual secrets data being completely managed by the users within their clusters.  

[Here](https://docs.google.com/presentation/d/1wymQAHUHhCOz4q-1aOlKU5paLv4GO0yFi3tVcOjeL7M/edit#slide=id.g11b5a925a28_0_0) is a visual representation of how an experiment pod consumes the secret on the chaos infrastructure side.

[Here](https://developer.harness.io/docs/chaos-engineering/chaos-faults/aws/ec2-cpu-hog#prerequisites) is an example of AWS access information being fed to the experiment pods via Kubernetes secret.

### Blast radius control using permissions 

> Depending upon (a) the infrastructure where chaos experiments are executed, and (b) desired list of chaos experiments, the permissions used can be modified (reduced) to create a lower blast radius and impact from a security perspective. This applies both to Kubernetes-based as well as non-Kubernetes chaos. 

Reviewer: ```suggestion
Depending on the infrastructure on which chaos experiments are executed and the desired list of chaos experiments, the permissions used can be modified (reduced) to create a lower blast radius and impact from a security perspective. This applies to both Kubernetes-based and non-Kubernetes chaos. 
```
There is some ambiguity in the first sentence. Does it imply that the infrastructures and experiments in use determine whether or not users can modify permissions (for some infrastructures and experiments, they can; for others, they cannot)? The "depending upon" is making it a little unclear, and even if users understand it correctly, it won't be on the first pass. 

I think what is meant is this: You can fine-tune permissions to suit specific infrastructures and experiments if you want to reduce the blast radius and impact from a security perspective. 

Does that simplify the statement somewhat? When we use "depending on", we need to be clear.

> In the case of the Kubernetes chaos, it is achieved through [service accounts](https://docs.google.com/document/d/1uB7BVFDdD8NfaT97MTj0qrF0eE3rnQOmpOOi4AqCzWE/edit#heading=h.p793kkfaj1zi) mapped to custom roles instead of the default ones mentioned in the [Kubernetes roles for chaos infrastructure](#kubernetes-roles-for-chaos-infrastructure). For non-Kubernetes chaos, it is achieved via cloud-specific role definitions (for example, IAM) mapped to the user account.  

Reviewer: Does "default ones" refer to "default service accounts" or "default roles"? Replace "ones" accordingly.

```suggestion
In the case of the Kubernetes chaos, it is achieved through [service accounts](https://docs.google.com/document/d/1uB7BVFDdD8NfaT97MTj0qrF0eE3rnQOmpOOi4AqCzWE/edit#heading=h.p793kkfaj1zi) mapped to custom roles instead of the default ones mentioned in [Kubernetes roles for chaos infrastructure](#kubernetes-roles-for-chaos-infrastructure). For non-Kubernetes chaos, it is achieved through cloud-specific role definitions (for example, IAM) mapped to the user account.  
```
Also, please clarify what "it" refers to in both statements. Thanks.

> Each fault in the enterprise ChaosHub publishes the permissions needed in order to execute it and users are free to tune their roles accordingly. Also available are common permission templates that work as subsets/supersets for a specific category of experiments. For example, here are recommended roles for all [AWS resource faults](https://developer.harness.io/docs/chaos-engineering/chaos-faults/aws/policy-for-all-aws-faults). 

Reviewer: ```suggestion
Every fault in the Enterprise ChaosHub publishes the permissions that users need to execute the fault. Users can tune their roles. Also available are common permission templates that work as subsets/supersets for a specific category of experiments. For example, for information about recommended roles for AWS resource faults, go to [AWS resource faults](https://developer.harness.io/docs/chaos-engineering/chaos-faults/aws/policy-for-all-aws-faults). 
```
(Modified the statement that includes the reference to AWS resource faults because the link was not applied to the word "here", and that made me look below this paragraph for the roles.)

### Privileged execution of chaos faults

> The deployments that comprise of the chaos infrastructure and the transient experiment pods launched to inject faults use the [in-cluster config](https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/) to make Kubernetes API calls, and thereby auto-mount service token secrets. The execution happens through non-root users with containers running secure base images. 

Reviewer: ```suggestion
The deployments that comprise the chaos infrastructure and the transient experiment pods launched to inject faults use the [in-cluster configuration](https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/) to make Kubernetes API calls and thereby auto-mount service token secrets. The execution happens through non-root users with containers running secure base images. 
```

> Some faults (mainly, the pod network and stress faults) necessitate container-runtime specific operations which necessitate privilege escalation, such as entering the network and pid namespaces, manipulating cgroup etc. In these cases, some of the pods are designed to run with privileged containers and root users. These pods also mount the runtime-specific socket files from the underlying host. However, it is important to note that such pods are short-lived (they exist for the duration of chaos) and can be run only if the users equip the serviceaccounts with access to the right security policy. 

Reviewer: ```suggestion
Some faults (mainly, the pod network and stress faults) necessitate container-runtime—specific operations such as entering the network and pid namespaces. These operations in turn necessitate privilege escalation, manipulating the cgroup, and so on. In these cases, some of the pods are designed to run with privileged containers and root users. These pods also mount the runtime-specific socket files from the underlying host. However, it is important to note that such pods are short-lived (they exist for the duration of chaos) and can be run only if the users equip the serviceaccounts with access to the right security policy. 
```

> Harness recommends security policy templates ([PSP](https://kubernetes.io/docs/concepts/security/pod-security-policy/), [OpenShift SCC](https://docs.openshift.com/container-platform/3.11/admin_guide/manage_scc.html), [Kyverno](https://kyverno.io/policies/)) to enable the execution of such experiments. 

Reviewer: ```suggestion
To enable the execution of such experiments, Harness recommends the security policy templates ([PSP](https://kubernetes.io/docs/concepts/security/pod-security-policy/), [OpenShift SCC](https://docs.openshift.com/container-platform/3.11/admin_guide/manage_scc.html), and [Kyverno](https://kyverno.io/policies/)). 
```
